File: SDKs/amazonaws.com/personalize/2018-05-22/go/pkg/models/operations/createeventtracker.go
package operations

import (
	"openapi/pkg/models/shared"
)

type CreateEventTrackerXAmzTargetEnum string

const (
	CreateEventTrackerXAmzTargetEnumAmazonPersonalizeCreateEventTracker CreateEventTrackerXAmzTargetEnum = "AmazonPersonalize.CreateEventTracker"
)

type CreateEventTrackerHeaders struct {
	XAmzAlgorithm     *string                          `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                          `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                          `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                          `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                          `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                          `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                          `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        CreateEventTrackerXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type CreateEventTrackerRequest struct {
	Headers CreateEventTrackerHeaders
	Request shared.CreateEventTrackerRequest `request:"mediaType=application/json"`
}

type CreateEventTrackerResponse struct {
	ContentType                    string
	CreateEventTrackerResponse     *shared.CreateEventTrackerResponse
	InvalidInputException          *interface{}
	LimitExceededException         *interface{}
	ResourceAlreadyExistsException *interface{}
	ResourceInUseException         *interface{}
	ResourceNotFoundException      *interface{}
	StatusCode                     int64
}

File: SDKs/amazonaws.com/personalize/2018-05-22/typescript/src/sdk/models/operations/createeventtracker.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum CreateEventTrackerXAmzTargetEnum {
    AmazonPersonalizeCreateEventTracker = "AmazonPersonalize.CreateEventTracker"
}


export class CreateEventTrackerHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: CreateEventTrackerXAmzTargetEnum;
}


export class CreateEventTrackerRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: CreateEventTrackerHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.CreateEventTrackerRequest;
}


export class CreateEventTrackerResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  createEventTrackerResponse?: shared.CreateEventTrackerResponse;

  @SpeakeasyMetadata()
  invalidInputException?: any;

  @SpeakeasyMetadata()
  limitExceededException?: any;

  @SpeakeasyMetadata()
  resourceAlreadyExistsException?: any;

  @SpeakeasyMetadata()
  resourceInUseException?: any;

  @SpeakeasyMetadata()
  resourceNotFoundException?: any;

  @SpeakeasyMetadata()
  statusCode: number;
}
